packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -88,7 +88,12 @@ export default util.createRule<Options, MessageIds>({
 
       if (type.isIntersection()) {
         const types = type.types.map(getBaseTypeOfLiteralType);
-        return types.some(value => value === 'string') ? 'string' : 'invalid';
+
+        if (types.some(value => value === 'string')) return 'string';
```

Comment:
Quick comment not a full review.
Please don't use single line ifs.
Add curlies and a body.